packages/core/src/shipping/strategies/stripe-upe/stripe-upe-shipping-strategy.ts

```diff
@@ -0,0 +1,113 @@
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId, { params: { method: 'card' } }));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow('card', methodId);
+        const { initializationData: { stripePublishableKey, stripeConnectedAccount } } = paymentMethod;
+
+        if (!paymentMethod || !paymentMethod.initializationData.stripePublishableKey || !paymentMethod.clientToken) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+        this._stripeUPEClient = await this._stripeUPEScriptLoader.getStripeClient(stripePublishableKey, stripeConnectedAccount);
+
+        let appearance: StripeUPEAppearanceOptions;
+        const styles = getStyles && getStyles();
+        console.log(styles);
+        if (styles) {
+            appearance = {
+                variables: {
+                    colorPrimary: styles.fieldInnerShadow,
+                    colorBackground: styles.fieldBackground,
+                    colorText: styles.labelText,
+                    colorDanger: styles.fieldErrorText,
+                    colorTextSecondary: styles.labelText,
+                    colorTextPlaceholder: styles.fieldPlaceholderText,
+                    spacingUnit: '4px',
+                    borderRadius: '4px',
+                },
+                rules: {
+                    '.Input': {
+                        borderColor: styles.fieldBorder,
+                        color: styles.fieldText,
+                        boxShadow: styles.fieldInnerShadow,
+                    },
+                }
+            };
+        } else {
+            appearance = {
+                variables: {
+                    spacingUnit: '4px',
+                    borderRadius: '4px',
+                }
+            };
+        }
+
+        this._stripeElements = this._stripeUPEScriptLoader.getElements(this._stripeUPEClient, {
+                clientSecret: paymentMethod.clientToken,
+                appearance: appearance,
+            });
+
+        const shippingAddressElement = this._stripeElements.getElement(StripeElementType.SHIPPING) ||
+            this._stripeElements.create(StripeElementType.SHIPPING, {allowedCountries: [availableCountries]});
+
+        shippingAddressElement.on('change', (event: StripeEventType) => {
+            if (!('isNewAddress' in event )) {
+                throw new MissingDataError(MissingDataErrorType.MissingCustomer);
+            } else {
+                if ((event.complete || event.isNewAddress) && onChangeShipping && typeof onChangeShipping === 'function') {
+                    if (this.sendData) {
+                        clearTimeout(this.sendData)
+                    }
+
+                    this.sendData = setTimeout(() => {
+                        onChangeShipping(event)
+                    }, 1000);
```

Comment:
Do we need a timeout here?

Reply:
I had to simulate the behavior of a debounce since for some strange reason it did not work as expected.

The timeout is being implemented because Stripe notifies us when the form is full but the user could continue writing, and that would send information many times, slowing down the shipping process. to avoid that, we decided to send information less often.